### ISSUE TITLE
Add support for thread-pool cpulist

### ISSUE THREAD

**Issue by sheinbergon** [RediSearch/RediSearch#1989]
Hello

We're using redisearch 2.0.6 with redis 6.0.11

Redis 6.x introduced fine grained control on various thread groups affinity settings. 
I believe redisearch  spawns a dedicated thread pool (when running  `pidstat` I can see a thread named `thread-pool-0`).

That name links back to [thpool.c](https://github.com/RediSearch/RediSearch/blob/master/src/dep/thpool/thpool.c)

Yet, that thread does not adhere to any affinity constraints. It'd be great if there was a way to ensure compute resource separation, as multi core instances are quite common these days, 

maybe something in the form of `THREAD_POOL_CPULIST 0,1-4` that would adhere to redis/taskset configuration standards

**Comment by sheinbergon** [comment]
@gkorland would you be willing to accept a PR?

**Comment by MeirShpilraien** [comment]
@sheinbergon we will be happy to accept a PR. I believe we can use the issue to define and scope what we want to do. IIUC we want a new configuration to define CPU's we want to run the threads on, and also we need another configuration that defines CPU to put the fork process created by forkgc. This configuration should be set on start time and should not be allowed to update on runtime. Let me know what you think. If you want we can talk privately on discord if you have any specific questions about the code.

**Comment by sheinbergon** [comment]
Yep, that's about it. You're right about the spec/scope. I'll see what I can do about it :)

**Comment by sheinbergon** [comment]
It should also only apply on Linux/*BSD distros, as macos' CPU affinity isn't really a thing a no one running a redis setup on windows should care about performance tweaking at this level.
